### PR TITLE
[WFLY-13023] Upgrade MicroProfile Config 1.4

### PR DESCRIPTION
**Pull request by jmesnil**
JIRA: https://issues.redhat.com/browse/WFLY-13023

Signed-off-by: Jeff Mesnil <jmesnil@redhat.com>

### CODE REVIEW

File: microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+=== Non-Requirements
+
+== Implementation Plan
+
+== Test Plan
+
+* Run the WildFly integration basic test suite and checks there are no failures related to MicroProfle Config tests.
+* Run the `microprofile-tck/config` module from WildFly integration test suite and checks that there are no failures.
+* Some tests that relates to the https://github.com/eclipse/microprofile-config/issues/446[inconsistent behaviour for empty/missing config property (#446)] are excluded as smallrye-config implementation does not pass the TCK tests that are challenged by this issue.
```
Comment (marekkopecky) [inline microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc]: > smallrye-config implementation does not pass the TCK tests

@jmesnil Can you please share more details about this? Which tests doesn't pass? I don't see any ignored tests in WF TS

> Some tests that relates to the https://github.com/eclipse/microprofile-config/issues/446[inconsistent behaviour for empty/missing config property (#446)] are excluded ...

Can you please share list of these tests?

> https://github.com/eclipse/microprofile-config/issues/446

I'm looking to this issue and it seems that there is no conclusion about future behaviour.

File: microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
+= Upgrade MicroProfile Config to 1.4
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          microprofile,config,observability
+:idprefix:
+:idseparator:       -
+
+== Overview
```
Comment (marekkopecky) [inline microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc]: Can you please summarize main differences between `1.3` and `1.4` MP Config versions in this Overview section?

File: microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc (hunk 1 above)
Comment (jmesnil) [reply microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc]: @marekkopecky, test excluded from the MP config TCK are configured in https://github.com/wildfly/wildfly/blob/3af5d5438457bc2697ffc59118e7d5ce247f875d/testsuite/integration/microprofile-tck/config/tck-suite.xml#L11

The discussion in https://github.com/eclipse/microprofile-config/issues/446 is ongoing will not be resolved before MP Config 2.0 is released

File: microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc (hunk 2 above)
Comment (jmesnil) [reply microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc]: I'll add a link to the release notes

File: microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+* Upgrade MicroProfile Config to 1.4
+* Upgrade smallrye-config to the release that implements MicroProfile Config 1.4
+* Run any MicroProfile Health test in WildFly integration test suite
+* Run the Eclipse MicroProfile Health TCK without failures
+
+=== Non-Requirements
+
+== Implementation Plan
+
+== Test Plan
```
Comment (marekkopecky) [inline microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc]: @jmesnil What is important to have test coverage in your PoV? Currently, this section doesn't contain anything about new tests.

File: microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc (hunk 2 above)
Comment (jmesnil) [reply microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc]: done

File: microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc (hunk 3 above)
Comment (jmesnil) [reply microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc]: I mentioned that spec changes are covered by the MP Config TCK and do not require new tests in WildFly

File: microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc (hunk 1 above)
Comment (marekkopecky) [reply microprofile/WFLY-13023_upgrade_microprofile-config_1.4.adoc]: I check WF TS and SmallRye TS (1.6.1). Excluded TCK tests fails in both these TSs

**Comment by marekkopecky** [comment]
@jmesnil Thank you for your updates.

**Pull request merged**